### ISSUE TITLE
[AI] - Finetune our finetuned model on the sample DFDC videos

### ISSUE THREAD

**Issue by Crownedprinz** [vbodhani11/Deepfake-Detector#49]
## Overview
This task provides a complete pipeline for fine-tuning a our saved model on deepfake detection using face crops extracted from video data.

### Step 1: Data Splitting (Prerequisite)
**Before using this notebook**, you must first split your video data using the `split_chunks_00_into_train_val_test_john.ipynb` notebook:
- Use the **exact same splitting method** (80/10/10 train/val/test ratio with random_state=42)
- The split notebook organizes videos into the following structure:
  ```
  organized/
    train/
      real_videos/*.mp4
      fake_videos/*.mp4
    val/
      real_videos/*.mp4
      fake_videos/*.mp4
    test/
      real_videos/*.mp4
      fake_videos/*.mp4
  ```

### Step 2: Update Configuration
After splitting your data, update the folder paths in the **User Configuration** section (Cell 0) to point to your sample data:
- Change `data_path` to your organized data directory
- Update `frames_base_dir`, `crops_base_dir`, and `run_base_dir` as needed

### Step 3: Run Fine-tuning Pipeline
This notebook will:
1. Extract frames from videos (3 fps)
2. Detect and crop faces from frames using MediaPipe
3. Fine-tune Xception model in two phases:
   - **Phase 1**: Freeze backbone, train classifier head only
   - **Phase 2**: Unfreeze all layers, fine-tune entire model
4. Track and visualize:
   - Validation loss over epochs
   - Confusion matrix for validation set
5. Evaluate on test set with metrics and confusion matrix

## Expected Outputs
- Trained model checkpoints (best and last)
- Validation loss graph showing training progress
- Confusion matrix visualization
- Test set evaluation metrics (Accuracy, F1, AUC, AP)
